odc/stats/plugins/tcw_percentiles.py

```diff
@@ -76,10 +112,18 @@ def native_transform(self, xx):
         """
         Loads data in its native projection.
         """
-        bad = enum_to_bool(
-            xx["fmask"], ("nodata", "cloud", "shadow")
-        )  # a pixel is bad if any of the cloud, shadow, or no-data value
-        bad |= xx["nbart_contiguity"] == 0  # or the nbart contiguity bit is 0
+        mask = xx["fmask"]
+        nodata = enum_to_bool(mask, ("nodata",))
+        non_contiguent = xx["nbart_contiguity"] == 0
+        bad = nodata | non_contiguent
+        # Now exclude cloud and cloud shadow (including buffered pixels)
+        if self.cloud_filters is not None:
+            for cloud_class, c_filter in self.cloud_filters.items():
+                cloud_mask = enum_to_bool(mask, (cloud_class,))
+                cloud_mask_buffered = mask_cleanup(cloud_mask, mask_filters=c_filter)
+                bad = cloud_mask_buffered | bad
+        else:
+            _log.info("There is no cloud/shadow buffering.")
```

Comment:
It won't do cloud/shadow mask at all if there is no dilation. We'll still need cloud/shadow mask as what they're if it's the case.

Reply:
Oh yeah, good catch.  I had changed the default behaviour!  Fixed :)

Reply:
`self.cloud_filters` won't be `None` since it was set `{}` at line 39

Reply:
I'll approve it. Functionality wise, it's ok. Coding wise, it's slightly messy and not ideal for dask graphs.

Note to myself: do some refactor when/if any change is required.